### PR TITLE
use a lazyily initialized stream for `node:crypto` `createHash`

### PR DESCRIPTION
**Pull request by paperclover**
```ts
createHash(algo).update(string).digest(format)
```

Will be over 3x faster, just shy of what our native `Bun.CryptoHasher` does. This is done by lazily initializing the `stream.Transform` only if the user decides to use the `Hash` as a stream, meaning the above case only calls CryptoHasher, giving it that performance boost.
 
image

The exact way this is done is by extending `Transform` but not calling ""`super`"" in the constructor (it's an es5 class so this is allowed). Then there are a ton of getters and if any are triggered, we call the `Transform` constructor on our current instance and replace the prototype.

Addresses #2190 but not the issue it's talking about, but it does solve performance issues for what it's testing for.

### CODE REVIEW

File: bench/snippets/crypto-2190.cjs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// https://github.com/oven-sh/bun/issues/2190
+const Benchmark = require("benchmark");
```
Comment (Jarred-Sumner) [inline bench/snippets/crypto-2190.cjs]: can you port this to use mitata

later when we add a website for our snippets we can turn it into JSON

Comment (paperclover) [reply bench/snippets/crypto-2190.cjs]: yeah i'll go ahead and do that

File: src/bun.js/crypto.exports.js [hunk 2]
```diff
@@ -1134,64 +1134,148 @@ var require_browser2 = __commonJS({
...
+      "writableBuffer",
+      "writableCorked",
+      "writableEnded",
+      "writableFinished",
+      "writableHighWaterMark",
+      "writableLength",
+      "writableNeedDrain",
+      "writableObjectMode",
+      "write",
+    ];
```
Comment (Jarred-Sumner) [inline src/bun.js/crypto.exports.js:1264]: `_writableState` might be needed too

**Comment by paperclover** [comment]
We are now testing stream cases. not too in-depth but its better than nothing.
image
We also have two tests making sure Hash looks like it extends Transform even though it doesn't exactly do this.

**Pull request merged**